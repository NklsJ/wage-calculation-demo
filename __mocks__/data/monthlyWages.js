
export default [
  {
    month: 1,
    year: 2015,
    workers: [
      {
        id: 1,
        fullname: 'Foo Bar',
        days: [
          {
            date: '1.1.2015',
            eveningHours: 0,
            hours: 8,
            overtimeHours: 0,
            shifts: [
              {
                start: '8:00',
                end: '16:00',
              },
            ],
            wage: 34,
          },
          {
            date: '2.1.2015',
            eveningHours: 0,
            hours: 10,
            overtimeHours: 2,
            shifts: [
              {
                start: '6:00',
                end: '10:00',
              },
              {
                start: '12:00',
                end: '18:00',
              },
            ],
            wage: 44.63,
          },
        ],
      },
      {
        id: 2,
        fullname: 'Bar Baz',
        days: [
          {
            date: '1.1.2015',
            eveningHours: 0,
            hours: 8,
            overtimeHours: 0,
            shifts: [
              {
                start: '8:00',
                end: '16:00',
              },
            ],
            wage: 34,
          },
          {
            date: '2.1.2015',
            eveningHours: 6,
            hours: 10,
            overtimeHours: 2,
            shifts: [
              {
                start: '00:00',
                end: '10:00',
              },
            ],
            wage: 52.13,
          },
        ],
      },
      {
        id: 3,
        fullname: 'Asd Das',
        days: [
          {
            date: '1.1.2015',
            eveningHours: 2,
            hours: 12,
            overtimeHours: 4,
            shifts: [
              {
                start: '4:00',
                end: '16:00',
              },
            ],
            wage: 58.81,
          },
          {
            date: '3.1.2015',
            eveningHours: 8,
            hours: 12,
            overtimeHours: 4,
            shifts: [
              {
                start: '20:00',
                end: '21:00',
              },
              {
                start: '23:00',
                end: '10:00',
              },
            ],
            wage: 66.31,
          },
        ],
      },
    ],
  },
]
