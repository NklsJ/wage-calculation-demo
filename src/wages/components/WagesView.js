// @flow
import React from 'react'
import WagesTable from './WagesTable'
import {type MonthlyWages} from '../types'

type Props = {
  monthlyWages: MonthlyWages,
}

const WagesView = ({monthlyWages}: Props) => {

  return (
    <div>
      {monthlyWages.map((item, index) => {

        return (
          <div key={index}>
            <h3>{`Monthly wages: ${item.month}/${item.year}`}</h3>

            <WagesTable workers={item.workers} />
          </div>
        )
      })
      }
    </div>
  )
}

export default WagesView
