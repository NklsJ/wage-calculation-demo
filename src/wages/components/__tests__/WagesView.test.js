import React from 'react'
import {shallow} from 'enzyme'

import WagesView from '../WagesView'
import monthlyWages from '../../../../__mocks__/data/monthlyWages'

describe('WagesView component: ', () => {

  it('Should render correctly', () => {
    const component = shallow(<WagesView monthlyWages={monthlyWages} />)
    expect(component).toMatchSnapshot()
  })

})
