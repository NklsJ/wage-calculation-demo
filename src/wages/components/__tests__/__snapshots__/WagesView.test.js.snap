// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WagesView component:  Should render correctly 1`] = `
<div>
  <div
    key="0"
  >
    <h3>
      Monthly wages: 1/2015
    </h3>
    <WagesTable
      workers={
        Array [
          Object {
            "days": Array [
              Object {
                "date": "1.1.2015",
                "eveningHours": 0,
                "hours": 8,
                "overtimeHours": 0,
                "wage": 34,
              },
              Object {
                "date": "2.1.2015",
                "eveningHours": 0,
                "hours": 10,
                "overtimeHours": 2,
                "wage": 44.63,
              },
            ],
            "fullname": "Foo Bar",
            "id": 1,
          },
          Object {
            "days": Array [
              Object {
                "date": "1.1.2015",
                "eveningHours": 0,
                "hours": 8,
                "overtimeHours": 0,
                "wage": 34,
              },
              Object {
                "date": "2.1.2015",
                "eveningHours": 6,
                "hours": 10,
                "overtimeHours": 2,
                "wage": 52.13,
              },
            ],
            "fullname": "Bar Baz",
            "id": 2,
          },
          Object {
            "days": Array [
              Object {
                "date": "1.1.2015",
                "eveningHours": 2,
                "hours": 12,
                "overtimeHours": 4,
                "wage": 58.81,
              },
              Object {
                "date": "3.1.2015",
                "eveningHours": 8,
                "hours": 12,
                "overtimeHours": 4,
                "wage": 66.31,
              },
            ],
            "fullname": "Asd Das",
            "id": 3,
          },
        ]
      }
    />
  </div>
</div>
`;
